### PR TITLE
[Contributing] Fork and Action notifications

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to call out that creating a fork of the repository is a needed prerequisite and to advise that GitHub Actions should be manually disabled in forks.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -13,6 +9,7 @@
 - Install the latest version of git (https://git-scm.com/downloads)
 - Install [PowerShell](https://docs.microsoft.com/powershell/scripting/install/installing-powershell), version 6 or higher, if you plan to make public API changes or are working with generated code snippets.
 - Install [NodeJS](https://nodejs.org/) (16.x.x) if you plan to use [C# code generation](https://github.com/Azure/autorest.csharp).
+- [Fork the repository](https://docs.github.com/en/get-started/quickstart/fork-a-repo); work will be done in your fork and a pull request should be opened agains the `main` branch of the Azure SDK for .NET repository when ready for review.
```
Comment (jsquire) [inline CONTRIBUTING.md:12]:
```suggestion
- [Fork the repository](https://docs.github.com/get-started/quickstart/fork-a-repo); work will be done in your fork and a pull request should be opened agains the `main` branch of the Azure SDK for .NET repository when ready for review.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,10 +1,6 @@
 # Contributing
 
-| Component            | Build Status                                                                                                                                                                                                          |
```
Comment (jsquire) [inline CONTRIBUTING.md:3]: Not sure why this was included in a contributing guide, but the badge stated "Never run" so it wasn't providing any value.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -13,6 +9,7 @@
 - Install the latest version of git (https://git-scm.com/downloads)
 - Install [PowerShell](https://docs.microsoft.com/powershell/scripting/install/installing-powershell), version 6 or higher, if you plan to make public API changes or are working with generated code snippets.
 - Install [NodeJS](https://nodejs.org/) (16.x.x) if you plan to use [C# code generation](https://github.com/Azure/autorest.csharp).
+- [Fork the repository](https://docs.github.com/get-started/quickstart/fork-a-repo); work will be done in your fork and a pull request should be opened against the `main` branch of the Azure SDK for .NET repository when ready for review.
```
Comment (annelo-msft) [inline CONTRIBUTING.md:12]: There's a section on forking the repo in the management-plane specific instructions: https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#standard-process

Does this conflict with that?

File: CONTRIBUTING.md (hunk 2 above)
Comment (annelo-msft) [reply CONTRIBUTING.md:3]: It looks like there was once one for Client Libraries, too! https://github.com/Azure/azure-sdk-for-net/pull/7052/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R6

File: CONTRIBUTING.md (hunk 3 above)
Comment (heaths) [inline CONTRIBUTING.md:12]: I'd go so far as to say a topic branch should be opened as well. You don't want your `origin/main` to differ from `upstream/main` - especially when we squash merge - or you're going to have a bad time. You could link to a sibling doc on GitHub: https://docs.github.com/en/get-started/quickstart/github-flow#create-a-branch (either that bookmark, or just the whole page).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -363,17 +362,16 @@ See [Data Plane Quick Start Tutorial](https://github.com/Azure/azure-sdk-for-net
...
-6. Generate the code. See [Generating Client Code](#generating-client-code) below.
-7. **MANDATORY**: Add or update tests for the newly generated code.
-8. Once added to the Azure SDK for .NET, build your local package using [client](#client-libraries) or [management](#management-libraries) library instructions shown in the above sections.
-9. For management libraries run `eng\scripts\Update-Mgmt-Yml.ps1` to update PR include paths in `eng\pipelines\mgmt.yml`
-10. A Pull request of your Azure SDK for .NET changes against **main** branch of the [Azure SDK for .NET](https://github.com/azure/azure-sdk-for-net)
-11. The pull requests will be reviewed and merged by the Azure SDK team
+1. Fork the [Azure REST API Specs](https://github.com/azure/azure-rest-api-specs) repository
+2. Create your Swagger specification for your HTTP API. For more information see [Introduction to Swagger - The World's Most Popular Framework for APIs](https://swagger.io)
+3. Install the latest version of AutoRest. For more info on getting started with AutoRest, see the [AutoRest repository](https://github.com/Azure/autorest)
+4. **RECOMMENDED**: Create a branch in your fork of the Azure SDK for .NET.
```
Comment (heaths) [inline CONTRIBUTING.md:368]: I would go so far as to say this is mandatory. See my comment about. When they try to sync `upstream/main` later, they're going to get all sorts of conflicts and, even if resolved, introduce a merge commit into their topic branch in some way that will likely cause dozens of codeowners to be added later.

**Review by heaths — APPROVED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (jsquire) [reply CONTRIBUTING.md]: I agree on that being the best practice, we just can't enforce it.   I updated to mark as mandatory, regardless.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -13,6 +9,7 @@
 - Install the latest version of git (https://git-scm.com/downloads)
 - Install [PowerShell](https://docs.microsoft.com/powershell/scripting/install/installing-powershell), version 6 or higher, if you plan to make public API changes or are working with generated code snippets.
 - Install [NodeJS](https://nodejs.org/) (16.x.x) if you plan to use [C# code generation](https://github.com/Azure/autorest.csharp).
+- [Fork the repository](https://docs.github.com/get-started/quickstart/fork-a-repo); work will be done on a topic branch in your fork and a pull request opened against the `main` branch of the Azure SDK for .NET repository when ready for review.
```
Comment (jsquire) [inline CONTRIBUTING.md:12]:
```suggestion
- [Fork the repository](https://docs.github.com/get-started/quickstart/fork-a-repo); work will be done on a [topic branch](https://docs.github.com/get-started/quickstart/github-flow#create-a-branch) in your fork and a [pull request opened](https://docs.github.com/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork) against the `main` branch of the Azure SDK for .NET repository when ready for review.
```

File: CONTRIBUTING.md (hunk 3 above)
Comment (jsquire) [reply CONTRIBUTING.md]: Added mention of a topic branch and linked in your suggested doc as well as one for opening PRs from a fork.

**Pull request merged**